src/Symfony/Component/Routing/Loader/XmlFileLoader.php

```diff
@@ -107,16 +107,20 @@ public function supports($resource, $type = null)
      */
     protected function parseRoute(RouteCollection $collection, \DOMElement $node, $path)
     {
-        if ('' === ($id = $node->getAttribute('id')) || !$node->hasAttribute('path')) {
-            throw new \InvalidArgumentException(sprintf('The <route> element in file "%s" must have an "id" and a "path" attribute.', $path));
+        if ('' === ($id = $node->getAttribute('id')) || (!$node->hasAttribute('path') && 'true' !== $node->getAttribute('empty-path'))) {
```

Comment:
an xml boolean can also be `1`

Reply:
let's make it accept only "true" (following you next comment)